src/_guides/language/effective-dart/design.md

```diff
@@ -1468,11 +1523,12 @@ Duration(
     int microseconds: 0});
 {% endprettify %}
 
-### AVOID mandatory parameters that permit nonce values.
+
+### AVOID mandatory parameters that permit sentinel values.
```

Comment:
Did you mean "special" here too? (I've used "sentinel" in the context of collections search, but in this context it seems strange.)

Reply:
I meant "sentinel", but it's not a perfect match. It's the best word I could come up with: https://en.wikipedia.org/wiki/Sentinel_value. "Nonce" isn't right.

Reply:
From Wikipedia (emphasis mine):

> **sentinel value** is a special value in the context of an algorithm which uses its presence as a _condition of termination_, typically in a loop or recursive algorithm

The value isn't being used "as a condition of termination".

Maybe the title needs to be reworked:

### AVOID implementing optional parameters as mandatory parameters

The opening paragraph then clearly explains what is meant by this.

@kwalrath - any thoughts on this too?

Reply:
I think that wording is a little too clever and likely to confuse. Changed it to:

> AVOID mandatory parameters that accept a special "no argument" value.